Salesforce Pardot destination

mohan-h opened the pull request:
Intial commit

mohan-h committed:
Salesforce Pardot destination

Intial commit

mohan-h committed:
Screengrab updated

mohan-h committed:
Corrections post connector functionality clarifications

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+---
+keywords: email;Email;e-mail;email destinations;salesforce;salesforce pardot;salesforce marketing cloud account engagement;salesforce pardot destination
```
minor, but from now on let's please drop the `keywords` metadata. historically, this was used to improve search results on our documentation website, but this is not needed anymore following some website improvements.

mohan-h replied:
Assume across all streams i.e. destination, Source and Extension. Please let me know if otherwise.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+# [!DNL Salesforce Pardot] connection
+
+[[!DNL Salesforce Pardot]](https://www.salesforce.com/products/marketing-cloud/overview/) *(now renamed as [!DNL Marketing Cloud Account Engagement])* allows you to capture and track leads by building templates, drag-and-drop builders; its powerful visual editor allows you to easily create custom landing pages, forms, and offers that attract potential leads for your sales teams. When someone enga ...
+
```
Suggestion for the overview -- I remember in the demo you briefly described how this destination is generally aimed at B2B use cases while other Salesforce destinations in our catalog are more aimed at B2C use cases. 

I think that information would be valuable here. Would you mind adding a blurb on that here?

mohan-h replied:
Added, also included a reference to the other connectors, if not required post back.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+
+## Prerequisites {#prerequisites}
+
+Refer to the sections below for any prerequisites that have to be setup / information that is to be gathered before working with the [!DNL Salesforce Pardot] destination.
```
Rephrasing suggestion to switch passive to active voice: 

Refer to the sections below for any prerequisites that you need to be set up in Experience Platform and Salesforce and for information that you need to gather before working with the [!DNL Salesforce Pardot] destination.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+>[!TIP]
+>
+>The [!DNL Salesforce Pardot] destination uses the selected segment names, to create custom prospect fields in [!DNL Salesforce Pardot], and update them with the Experience Platform segment status. The destination automatically replaces whitespace or special characters with the underscore `(_)` character as a separator to comply with [!DNL Salesforce Pardot] requirements.
+>However, as a recommendation, when creating a segment, do not include whitespace or special characters in the segment name. Instead, use the underscore `(_)` character as a separator. For example instead of naming your segment as `Salesforce Pardot Segment 1` use `Salesforce_Pardot_Segment_1`.
```
As discussed, I think that you can remove this suggestion, as we would expect perhaps customers to create segments that they reuse across multiple destinations.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+
+>[!TIP]
+>
+>The [!DNL Salesforce Pardot] destination uses the selected segment names, to create custom prospect fields in [!DNL Salesforce Pardot], and update them with the Experience Platform segment status. The destination automatically replaces whitespace or special characters with the underscore `(_)` character as a separator to comply with [!DNL Salesforce Pardot] requirements.
```
minor: please remove the comma before `to create custom prospect fields`

mohan-h replied:
As the callout has been removed, the recommendation is moot.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+
+#### Configure a connected app within [!DNL Salesforce] {#prerequisites-connected-app}
+
+You can follow the [!DNL Salesforce] documentation [here](https://help.salesforce.com/s/articleView?id=sf.pardot_sf_connector_setup_v2_parent.htm&type=5) or alternatively check the steps below.
```
I think an intro would be good here, like: 

To connect Experience Platform to Salesforce, you need to configure a connected app within [!DNL Salesforce]. (does this make sense?)

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+
+| Item | Type | Notes |
+---------|----------|---------|
+| Export frequency | **[!UICONTROL Batch]** | Batch destinations export files to downstream platforms in increments of three, six, eight, twelve, or twenty-four hours. Read more about [batch file-based destinations](/help/destinations/destination-types.md#file-based).|
```
I don't think this is correct. This destination does not export files to downstream platforms. This is an API-based integration.
I think we should reuse the same information in this doc as in the same section from the (API) Salesforce Marketing Cloud destination:

https://experienceleague.adobe.com/docs/experience-platform/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.html?lang=en#export-type-frequency

vgiurgiu replied:
I don't think this is correct. This destination does not export files to downstream platforms. This is an API-based integration.
I think we should reuse the same information in this doc as in the same section from the (API) Salesforce Marketing Cloud destination:

https://experienceleague.adobe.com/docs/experience-platform/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.html?lang=en#export-type-frequency

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+>
+>To activate data, you need the **[!UICONTROL Manage Destinations]**, **[!UICONTROL Activate Destinations]**, **[!UICONTROL View Profiles]**, and **[!UICONTROL View Segments]** [access control permissions](/help/access-control/home.md#permissions). Read the [access control overview](/help/access-control/ui/overview.md) or contact your product administrator to obtain the required permissions.
+
+Read [Activate audience data to batch profile export destinations](/help/destinations/ui/activate-batch-profile-destinations.md) for instructions on activating audience segments to this destination.
```
Same as a comment further above, this is not a batch profile export destination. It is an API-based streaming destination. Please reuse the content from this section:

https://experienceleague.adobe.com/docs/experience-platform/destinations/catalog/email-marketing/salesforce-marketing-cloud-exact-target.html?lang=en#activate

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+
+    * An example with the above mapping is shown below:
+    ![Platform UI screenshot example showing Target mappings.](../../assets/catalog/email-marketing/salesforce-pardot/mappings.png)
+
```
Can other attributes be mapped as well or is it only possible to map identities? 
e.g. can a user map first names, last names, addresses, birthdays, etc?

mohan-h replied:
Yes, other attributes can also be updated, have linked to the API's fields page, updated the screenshot and the mappings table to showcase additional fields.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+
+To validate that you have correctly set up the destination, follow the steps below:
+
+1. Navigate to one of the segments you had selected. Then Select the **[!DNL Activation data]** tab, The **[!UICONTROL Mapping ID]** column displays the name of the custom field which is generated within the [!DNL Salesforce Pardot Prospects] page.
```
Minor grammar comment: 

~Then~ Select the **[!DNL Activation data]** tab. (period instead of comma)

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-pardot.md:
```diff
@@ -0,0 +1,203 @@
+1. Log in to the [[!DNL Salesforce]](https://login.salesforce.com/) website. Then navigate to the **[!DNL Account Engagement]** > **[!DNL Prospects]** > **[!DNL Pardot Prospects]** page and check if the prospects from the segment have been added / updated. Alternatively you can also access [[!DNL Salesforce Pardot]](https://pi.pardot.com/) and access the **[!DNL Prospects]** page.
+![Salesforce UI screenshot showing the Prospects page.](../../assets/catalog/email-marketing/salesforce-pardot/salesforce-pardot-prospects.png)
+
+1. To check if the prospects have been updated, click on a prospect and verify if the custom prospect field has been updated with the Experience Platform segment status.
```
minor: let's use select instead of click

vgiurgiu reviewed (commented):
Few comments here and there. Thanks for the work, let me know if you have questions, and I'll get back to you with regard to the naming of the destination.

mohan-h committed:
Connector renamed and addressed review comments

mohan-h committed:
Update salesforce-marketing-cloud-account-engagement.md

The Authorization has changed from OAuth 2 with password to Oauth 2 with Authorization code accordingly the connected app section is removed.

mohan-h committed:
Added connected app details

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+Compared to [!DNL Salesforce Marketing Cloud Engagement] which is more oriented towards **B2C** marketing, [!DNL Marketing Cloud Account Engagement] is ideal for **B2B** use cases involving multiple departments and decision makers which require longer sales and decision cycles. In addition, you also maintain closer proximity and integration with your CRM to make appropriate sales and marketing de ...
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [[!DNL Salesforce Account Engagement API > Prospect Upsert by Email]](https://developer.salesforce.com/docs/marketing/pardot/guide/prospect-v5.html#prospect-upsert-by-email) endpoint, to add / update your leads after activating them within a new [!DNL Marketing Cloud Account Engagement] segment.
```
instead of the `/`, let's use:

to add or update your leads after activating them within a new [!DNL Marketing Cloud Account Engagement] segment.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+### Send emails to contacts for marketing campaigns {#use-case-send-emails}
+
+The marketing department of an online platform wants to broadcast an email based marketing campaign to a curated audience of B2B leads. The platform's marketing team can add new leads / update existing lead information through Adobe Experience Platform, build segments from their own offline data, and send these segments to [!DNL Marketing Cloud Account Engagement], which can then be used to send  ...
```
let's replace `/` with `or`

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+Your [!DNL Salesforce] account should have the [!DNL Salesforce] `Account Engagement Administrator role`. This is required to [create custom prospect fields](https://help.salesforce.com/s/articleView?id=sf.pardot_fields_create_custom_field.htm&type=5).
+
+Finally, your account should also be able to access the [!DNL Account Engagement Lightning App](https://help.salesforce.com/s/articleView?id=sf.pardot_lightning_enable.htm&type=5).
```
minor - missing brackets: 

`[[!DNL Account Engagement Lightning App]]`

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+### Guardrails {#guardrails}
+
+* Refer to the [!DNL Marketing Cloud Account Engagement] [rate limits](https://developer.salesforce.com/docs/marketing/pardot/guide/overview.html#rate-limits) which details the limits imposed by your plan and would also apply to the Experience Platform executions.
```
minor - no need for the bullet at the start of the paragraph

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+![Platform UI screenshot showing how to authenticate to Marketing Cloud Account Engagement.](../../assets/catalog/email-marketing/salesforce-marketing-cloud-account-engagement/authenticate-destination.png)
+
+Next, Select [!UICONTROL Allow] in the subsequent window to give permissions to the **Adobe Experience Platform** app to access your [!DNL Salesforce Marketing Cloud Account Engagement] account. *You will need to do this only once*. Note, the domain redirect shown in the window begins with an *adobe* or *infosys* prefix, both are valid. 
```
Wondering if this sentence is really needed: 

`Note, the domain redirect shown in the window begins with an *adobe* or *infosys* prefix, both are valid.`

mohan-h replied:
In the walkthrough it was mentioned to call out, for now have removed it.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+1. In the **[!UICONTROL Mapping]** step, select **[!UICONTROL Add new mapping]**. You will see a new mapping row on the screen.
+1. In the **[!UICONTROL Select source field]** window, choose the **[!UICONTROL Select attributes]** category and select the XDM attribute or choose the **[!UICONTROL Select identity namespace]** and select an identity.
+1. In the **[!UICONTROL Select target field]** window, choose the **[!UICONTROL Select identity namespace]** and select an identity or choose **[!UICONTROL Select custom attributes]** category and specify from the list of [!DNL Prospect API fields](https://developer.salesforce.com/docs/marketing/pardot/guide/prospect-v5.html#fields) from the available schema.
```
minor, missing brackets: 

`[[!DNL Prospect API fields]]`

vgiurgiu reviewed (commented):
Looking very good! Just some really minor comments. As always, let me know please if you have any questions.

mohan-h committed:
Updates to address comments from review

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+---
+# [!DNL Salesforce Marketing Cloud Account Engagement] connection
+
+[[!DNL Salesforce Marketing Cloud Account Engagement]](https://www.salesforce.com/products/marketing-cloud/overview/) *(formerly known as [!DNL Pardot])* allows you to capture and track leads by building templates, drag-and-drop builders; its powerful visual editor allows you to easily create custom landing pages, forms, and offers that attract potential leads for your sales teams. When someone e ...
```
Couple of points here: 

1. Let's please edit the first link to https://www.salesforce.com/products/marketing-cloud/marketing-automation/ which I see talks directly about this functionality.
2. On second reading, I think that having such a long paragraph like the one below copied from the Salesforce site is maybe not an ideal practice. Can we shorten it please to one or maximum two sentences and link to Salesforce for more information?

> [[!DNL Salesforce Marketing Cloud Account Engagement]](https://www.salesforce.com/products/marketing-cloud/overview/) *(formerly known as [!DNL Pardot])* allows you to capture and track leads by building templates, drag-and-drop builders; its powerful visual editor allows you to easily create custom landing pages, forms, and offers that attract potential leads for your sales teams. When someone engages with your forms, the lead is auto captured, and you get notified. Leads are also automatically scored and graded, so that those most likely to convert can be prioritized. Your forms can also trigger changes in lead scores upon conversion. [!DNL Marketing Cloud Account Engagement] enables you to design lead tracks for all stages of the pipeline for targeted market segments and customer groups through email drip campaigns, lead management with nurturing and scoring and campaign segmentation.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+Compared to [!DNL Salesforce Marketing Cloud Engagement] which is more oriented towards **B2C** marketing, [!DNL Marketing Cloud Account Engagement] is ideal for **B2B** use cases involving multiple departments and decision makers which require longer sales and decision cycles. In addition, you also maintain closer proximity and integration with your CRM to make appropriate sales and marketing de ...
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [[!DNL Salesforce Account Engagement API > Prospect Upsert by Email]](https://developer.salesforce.com/docs/marketing/pardot/guide/prospect-v5.html#prospect-upsert-by-email) endpoint, to add or update your leads after activating them within a new [!DNL Marketing Cloud Account Engagement] segment.
```
since this was an important topic with one of our customers, let's please bold the part `add or update your leads`

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+### Send emails to contacts for marketing campaigns {#use-case-send-emails}
+
+The marketing department of an online platform wants to broadcast an email based marketing campaign to a curated audience of B2B leads. The platform's marketing team can add new leads or update existing lead information through Adobe Experience Platform, build segments from their own offline data, and send these segments to [!DNL Marketing Cloud Account Engagement], which can then be used to send ...
```
really minor: email-based

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+## Additional resources {#additional-resources}
+
+* [!DNL Marketing Cloud Account Engagement] [API](https://developer.salesforce.com/docs/marketing/pardot/guide/overview.html)
```
really minor: [!DNL Marketing Cloud Account Engagement] [API documentation](https://developer.salesforce.com/docs/marketing/pardot/guide/overview.html)

vgiurgiu reviewed (commented):
some more comments, all relatively minor. after we work through these, we'll be good to go for publishing.

mohan-h committed:
Updates against review - 2

vgiurgiu reviewed (approved):
Thank you, @mohan-h ! Confirming that the documentation is now in a great state and ready to publish.

We'll publish it once we also approve the configuration of the destination itself.

vgiurgiu commented on help/destinations/catalog/email-marketing/salesforce-marketing-cloud-account-engagement.md:
```diff
@@ -0,0 +1,177 @@
+
+This [!DNL Adobe Experience Platform] [destination](/help/destinations/home.md) leverages the [[!DNL Salesforce Account Engagement API > Prospect Upsert by Email]](https://developer.salesforce.com/docs/marketing/pardot/guide/prospect-v5.html#prospect-upsert-by-email) endpoint, to **add or update your leads** after activating them within a new [!DNL Marketing Cloud Account Engagement] segment.
+
+[!DNL Marketing Cloud Account Engagement] uses [OAuth 2 with Authorization Code](/help/destinations/destination-sdk/functionality/authentication/oauth2-authentication.md#authorization-code) to communicate with the [!DNL Account Engagement] API. Instructions to authenticate to your [!DNL Marketing Cloud Account Engagement] instance are further below, in the [Authenticate to destination](#authentic ...
```
I missed this part in my reviews, but the link in this part is not truly relevant to customers: 

[!DNL Marketing Cloud Account Engagement] uses [OAuth 2 with Authorization Code](/help/destinations/destination-sdk/functionality/authentication/oauth2-authentication.md#authorization-code) to communicate with the [!DNL Account Engagement] API.

This is more of an internal implementation detail. I'll fix this before I publish the page, but just something to keep in mind for future docs.
